test/Microsoft.ML.Tests/OnnxSequenceTypeTest.cs

```diff
@@ -64,5 +82,21 @@ public void OnnxSequenceTypeWithColumnNameAttributeTest()
             }
 
         }
+
+        private static PredictionEngine<FloatInput, ProblematicOutputObj> CreatePredictorWithPronlematicOutputObj()
+        {
+            var onnxModelFilePath = Path.Combine(Directory.GetCurrentDirectory(), "zipmap", "TestZipMapString.onnx");
+
+            var ctx = new MLContext();
+            var model = PrepareModel(onnxModelFilePath, ctx);
+            return ctx.Model.CreatePredictionEngine<FloatInput, ProblematicOutputObj>(model);
+        }
+
+        [OnnxFact]
+        public void OnnxSequenceTypeWithouSpecifySequenceTypeTest()
+        {
+            Exception ex = Assert.Throws<Exception>(() => CreatePredictorWithPronlematicOutputObj());
+            Assert.Equal("Please specify sequence type when use OnnxSequenceType Attribute.", ex.Message);
+        }
```

Comment:
Would it be better to assert on the type of exception rather than the specific message?

Reply:
yes, the exception type is assert in previous line Assert.Throws<Exception>, after change to use ArgumentNullException, it should look like Assert.Throws<ArgumentNullException>(....)

---
In reply to: [330651709](https://github.com/dotnet/machinelearning/pull/4272#discussion_r330651709) [](ancestors = 330651709)